src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -60,18 +58,64 @@ private Pattern compileAllowlist(List<String> allowlist) {
         return Pattern.compile(joined);
     }
 
+    public String listener() {
+        return listener.toString();
+    }
+
+    static class Listener {
+
+        private static final Pattern PATTERN = Pattern.compile("http://\\[?([0-9a-zA-Z\\-%._:]*)]?:([0-9]+)");
+
+        final String host;
+        final int port;
+
+        public Listener(String host, int port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        public static Listener parseListener(String listener) {
+            Matcher matcher = PATTERN.matcher(listener);
+            if (matcher.matches()) {
+                String host = matcher.group(1);
+                int port = Integer.parseInt(matcher.group(2));
+                return new Listener(host, port);
+            } else {
+                throw new ConfigException(LISTENER_CONFIG, listener, "Listener be of format http://[host]:[port]");
+            }
+        }
+
+        @Override
+         public String toString() {
+            return "http://" + host + ":" + port;
+        }
+
+        @Override
+         public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Listener listener = (Listener) o;
+            return port == listener.port && Objects.equals(host, listener.host);
+        }
+
+        @Override
+         public int hashCode() {
+            return Objects.hash(host, port);
+        }
+    }
+
     @Override
     public String toString() {
         return "PrometheusMetricsReporterConfig{" +
                 "allowlist=" + allowlist +
-                ", port=" + port +
+                ", listener=" + listener +
                 '}';
     }
 
     public synchronized Optional<HTTPServer> startHttpServer() {
         try {
-            HTTPServer httpServer = new HTTPServer(port, true);
-            LOG.info("HTTP server started on port " + port);
+            HTTPServer httpServer = new HTTPServer(listener.host, listener.port);
```

Comment:
We still need to set `daemon` to `true`.

Reply:
Fixed